clarity/src/vm/functions/sequences.rs

```diff
@@ -383,3 +383,82 @@ pub fn special_slice(
     }?;
     Ok(sliced_seq)
 }
+
+pub fn special_replace_at(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let seq = eval(&args[0], env, context)?;
+    let seq_type = TypeSignature::type_of(&seq);
+    let (expected_elem_type, seq_is_list_type) =
+        if let TypeSignature::SequenceType(seq_subtype) = seq_type {
+            (seq_subtype.unit_type(), seq_subtype.is_list_type())
+        } else {
+            return Err(RuntimeErrorType::BadTypeConstruction.into());
+        };
+    let index = eval(&args[1], env, context)?;
+    let new_element = eval(&args[2], env, context)?;
+
+    if expected_elem_type != TypeSignature::NoType && !expected_elem_type.admits(&new_element) {
+        return Err(RuntimeErrorType::BadTypeConstruction.into());
+    }
```

Comment:
This check needs to change if `list` types should expand their type to fit the replaced-in element to something like:

```rust
if seq is list {
  TypeSignature::least_supertype(&expected_elem_type, &new_element_type)?;
} else {
  if !expected_elem_type.admits(&new_element_type) {
    return Err(CheckErrors::TypeValueError(expected_elem_type, new_element).into());
  }
}
```

 If lists aren't expanding at all, I think the check still needs to be updated to use the `CheckErrors` variant.